File: backend/src/models/Appointment.js
const mongoose = require('mongoose');

const appointmentSchema = new mongoose.Schema(
  {
    patient: { type: mongoose.Schema.Types.ObjectId, ref: 'Patient', required: true, index: true },
    doctor: { type: mongoose.Schema.Types.ObjectId, ref: 'Doctor', required: true, index: true },

    scheduledAt: { type: Date, required: true, index: true },
    duration: { type: Number, default: 15 },

    type: {
      type: String,
      enum: ['video', 'audio', 'in_person'],
      default: 'video',
    },

    status: {
      type: String,
      enum: [
        'scheduled',
        'confirmed',
        'in_progress',
        'completed',
        'cancelled',
        'no_show',
        'rescheduled',
      ],
      default: 'scheduled',
      index: true,
    },

    meetingLink: String,
    meetingId: String,
    meetingPassword: String,

    prescription: { type: mongoose.Schema.Types.ObjectId, ref: 'Prescription' },

    payment: {
      amount: { type: Number, default: 0 },
      method: { type: String, enum: ['upi', 'net_banking', 'cash', 'insurance'], default: 'cash' },
      transactionId: String,
      status: {
        type: String,
        enum: ['pending', 'completed', 'failed', 'refunded'],
        default: 'pending',
      },
      paidAt: Date,
    },

    cancellationReason: String,
    cancelledAt: Date,
    cancelledBy: { type: String, enum: ['patient', 'doctor', 'system'] },

    remindersSent: [
      {
        type: { type: String, enum: ['sms', 'push', 'email'] },
        sentAt: { type: Date, default: Date.now },
        status: { type: String, enum: ['sent', 'delivered', 'failed'] },
      },
    ],

    notes: {
      patient: String,
      doctor: String,
    },

    isSynced: { type: Boolean, default: false },
  },
  {
    timestamps: true,
    toJSON: { virtuals: true },
    toObject: { virtuals: true },
  }
);

appointmentSchema.virtual('isUpcoming').get(function () {
  return this.scheduledAt > new Date() && ['scheduled', 'confirmed'].includes(this.status);
});

appointmentSchema.virtual('canJoin').get(function () {
  if (this.status !== 'confirmed') return false;
  const diff = new Date(this.scheduledAt).getTime() - Date.now();
  return diff <= 15 * 60 * 1000 && diff >= -30 * 60 * 1000;
});

appointmentSchema.index({ patient: 1, scheduledAt: 1 });
appointmentSchema.index({ doctor: 1, scheduledAt: 1 });
appointmentSchema.index({ status: 1, scheduledAt: 1 });

module.exports = mongoose.model('Appointment', appointmentSchema);

File: symptom_checker/lib/core/models/appointment.dart
import 'package:json_annotation/json_annotation.dart';
import 'doctor.dart';

part 'appointment.g.dart';

enum AppointmentStatus {
  scheduled,
  confirmed,
  inProgress,
  completed,
  cancelled,
  noShow,
  rescheduled
}

enum AppointmentType {
  video,
  audio,
  inPerson
}

@JsonSerializable()
class Appointment {
  final String id;
  final String patientId;
  final String doctorId;
  final Doctor? doctor;
  final DateTime scheduledAt;
  final Duration duration;
  final AppointmentType type;
  final AppointmentStatus status;
  final String? meetingLink;
  final String? meetingId;
  final String? prescriptionId;
  final String? paymentId;
  final double amount;
  final String paymentMethod;
  final bool paymentCompleted;
  final String? cancellationReason;
  final DateTime? cancelledAt;
  final List<String> reminderSent;
  final String notes;
  final DateTime createdAt;
  final DateTime updatedAt;
  final bool isSynced;

  const Appointment({
    required this.id,
    required this.patientId,
    required this.doctorId,
    this.doctor,
    required this.scheduledAt,
    this.duration = const Duration(minutes: 15),
    required this.type,
    this.status = AppointmentStatus.scheduled,
    this.meetingLink,
    this.meetingId,
    this.prescriptionId,
    this.paymentId,
    this.amount = 0,
    this.paymentMethod = 'cash',
    this.paymentCompleted = false,
    this.cancellationReason,
    this.cancelledAt,
    this.reminderSent = const [],
    this.notes = '',
    required this.createdAt,
    required this.updatedAt,
    this.isSynced = false,
  });

  factory Appointment.fromJson(Map<String, dynamic> json) => _$AppointmentFromJson(json);
  Map<String, dynamic> toJson() => _$AppointmentToJson(this);

  String getStatusLabel(String locale) {
    final labels = {
      AppointmentStatus.scheduled: {'en': 'Scheduled', 'hi': 'निर्धारित', 'te': 'నిర్ధారిత', 'ta': 'திட்டமிடப்பட்ட', 'mr': 'नियोजित'},
      AppointmentStatus.confirmed: {'en': 'Confirmed', 'hi': 'पुष्टि', 'te': 'నిఖిల', 'ta': 'உறுதிப்படுத்தப்பட்டது', 'mr': 'पुष्टीकरण'},
      AppointmentStatus.inProgress: {'en': 'In Progress', 'hi': 'चल रहा है', 'te': 'ప్రગతిలో', 'ta': 'நடைபெறുകின்றது', 'mr': 'प्रगतीत'},
      AppointmentStatus.completed: {'en': 'Completed', 'hi': 'पूर्ण', 'te': 'పూర్తి', 'ta': 'முடிந்தது', 'mr': 'पूर्ण'},
      AppointmentStatus.cancelled: {'en': 'Cancelled', 'hi': 'रद्द', 'te': 'రద్దు', 'ta': 'ரத்து', 'mr': 'रद्द'},
      AppointmentStatus.noShow: {'en': 'No Show', 'hi': 'नहीं आए', 'te': 'రheimer', 'ta': 'வரவில்லை', 'mr': 'येत नाही'},
      AppointmentStatus.rescheduled: {'en': 'Rescheduled', 'hi': 'पुनर्निर्धारित', 'te': 'పునఃనిర్ధారిత', 'ta': 'மீண்டும் திட்டமிடப்பட்டது', 'mr': 'पुन्हा नियोजित'},
    };
    return labels[status]?[locale] ?? labels[status]?['en'] ?? status.name;
  }

  bool get isUpcoming => scheduledAt.isAfter(DateTime.now());
  bool get canJoin => status == AppointmentStatus.confirmed && 
      scheduledAt.difference(DateTime.now()).inMinutes <= 15;
}